flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamGraphGenerator.java

```diff
@@ -244,19 +245,30 @@ private void configureStreamGraph(final StreamGraph graph) {
 	}
 
 	private boolean shouldExecuteInBatchMode(final RuntimeExecutionMode configuredMode) {
+		final boolean existsUnboundedSource = existsContinuousSource();
+
+		if (configuredMode == RuntimeExecutionMode.BATCH && existsUnboundedSource) {
```

Comment:
Was also not decided on that. I went with `argument` because it is something set by the user. But it may make sense to put `state` because `argument` seems more mild.

Reply:
In similar cases I went with `state` because it is not obvious which `argument` on which method is illegal. Usually I'd use `argument` if it is thrown explicitly from a method that is called by the recipient of the exception. But I do agree the `state` is also not a perfect solution as it is because of a user setting.